extensions/blocks/calendly/edit.js

```diff
@@ -65,6 +69,25 @@ export default function CalendlyEdit( props ) {
 			</>
 		);
 
+	useEffect( () => {
+		if ( ! url || CALENDLY_EXAMPLE_URL === url || 'link' === style ) {
+			return;
+		}
+		setResolveUrl( true );
+		apiFetch( { path: `/wpcom/v2/resolve-redirect/${ url }` } ).then(
```

Comment:
Now that we rely on `wpcom/v2/resolve-redirect` here, in the Pinterest block, and in the Eventbrite block, would it make sense / be possible to extract this into a shared function maybe? What do you think?

Reply:
Good thinking!
I'll update this PR by adding an utility function, and then will take care of the other blocks separately to avoid bloating this.